content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,613 @@
+
+| Model (Country) | LSTM one layer | Bidirectional bi-LSTM |
+|:---------------:|:--------------:|:------------------:|
+|    Loss (US)    |     0.6176     |       **0.3078**   |
+|  Accuracy (US)  |     84.7396    |       **91.8220**  |
+|    Loss (UK)    |     0.4368     |       **0.1571**   |
+|  Accuracy (UK)  |     86.2543    |       **95.6840**  |
+
+
+##### The Second and Third Test
+
+Now let's tests for Russia and Mexico.
+
+```python
+ru_loader = DataLoader(ru_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(ru_loader)
+exp_bi_lstm.test(ru_loader)
+
+mx_loader = DataLoader(mx_data, batch_size=batch_size, collate_fn=pad_collate_fn)
+exp.test(mx_loader)
+exp_bi_lstm.test(mx_loader)
+```
+
+The next table presents the results of both models for the two countries tested. We see that the first test 
+(RU) gives poorer results than those for Mexican addresses, even if these latter are written in a different structure and language. 
+This situation could be explained by both languages' roots; Spanish is closer to French than Russian is. 
+An interesting thing is that even in a *difficult* annotation context, both models perform relatively well. 
+It suggests that our models have really learned the *logic* of an address sequence. It could also mean that if 
+we train our model longer, we could potentially improve our results. Other modifications that could improve our models are discussed in the next and final section.
+
+| Model (Country) | LSTM one layer | Bidirectional bi-LSTM |
+|:---------------:|:--------------:|:------------------:|
+|    Loss (RU)    |   **2.5181**   |       4.6118       |
+|  Accuracy (RU)  |   **48.9820**  |       47.3185      |
+|    Loss (MX)    |     2.6786     |     **1.7147**     |
+|  Accuracy (MX)  |     50.2013    |     **63.5317**    |
+
+### Summary
+In summary, we found that using a bidirectional bi-LSTM seems to perform better on addresses not seen during training, including those coming from other countries. Still, the results for addresses from other countries are not as good as those for Canadian addresses (training dataset). A solution to this problem could be to train a model using all the
+data from all over the world. This approach was used by [Libpostal](https://github.com/openvenues/libpostal), which trained a 
+CRF over an impressive near `100` million addresses (yes, **100 million**). If you want to explore this avenue, the data is publicly available.
+
+We also explored the idea that the language has a negative impact on the results, since we use monolingual word embeddings (i.e. French), 
+which is *normal* considering that they were trained for a specific language. A possible solution to that problem is the use of subword embedding composed of sub-division of a word instead of the complete one. For example, a two characters window embeddings of `H1A1` would be the aggregate embeddings of the subword `H1`, `1A` and `A1`. 
```

Comment:
```suggestion
which is *normal* considering that they were trained for a specific language. A possible solution to that problem is the use of subword embeddings composed of sub-divisiona of a word instead of the complete one. For example, a two characters window embeddings of `H1A1` would be the aggregated embeddings of the subword `H1`, `1A` and `A1`. 
```